pkg/s3wrapper/client.go

```diff
@@ -473,52 +476,94 @@ func (c *S3Client) ListObjectsByPrefix(ctx context.Context, prefix string) ([]st
 
 func (c *S3Client) UploadBootFiles(ctx context.Context, openshiftVersion string) error {
 	rhcosImage, err := c.versionsHandler.GetRHCOSImage(openshiftVersion)
-
 	if err != nil {
 		return err
 	}
 
-	return c.uploadBootFiles(ctx, c.GetBaseIsoObject(openshiftVersion), rhcosImage)
+	baseIsoObject := c.GetBaseIsoObject(openshiftVersion)
+	minimalIsoObject := isoeditor.GetMinimalIsoObject(openshiftVersion)
+	return c.uploadBootFiles(ctx, baseIsoObject, minimalIsoObject, rhcosImage, openshiftVersion)
 }
 
-func (c *S3Client) uploadBootFiles(ctx context.Context, isoObjectName, isoURL string) error {
+func (c *S3Client) uploadBootFiles(ctx context.Context, isoObjectName, minimalIsoObject, isoURL, openshiftVersion string) error {
 	log := logutil.FromContext(ctx, c.log)
 
-	exist, err := c.DoAllBootFilesExist(ctx, isoObjectName)
+	baseExist, err := c.DoAllBootFilesExist(ctx, isoObjectName)
+	if err != nil {
+		return err
+	}
+	minimalExist, err := c.DoesIsoFileExist(ctx, minimalIsoObject)
```

Comment:
This is strange, why not just call `c.DoesPublicObjectExist(ctx, minimalIsoObject)` here?

Reply:
just wanted to keep convention and encapsulate it.. but sure, looks cleaner.